include/openthread-br/config.h

```diff
@@ -35,6 +35,10 @@
 
 #if defined(OTBR_CONFIG_FILE)
 #include OTBR_CONFIG_FILE
+#elif OTBR_USER_CONFIG_HEADER_ENABLE
+// This configuration header file should be provided by the user when
+// OTBR_USER_CONFIG_HEADER_ENABLE is defined to 1.
+#include "user-config.h"
```

Comment:
```suggestion
#include "ot-posix-user-config.h"
```

Reply:
My intention is to include `openthread-br/user-config.h`. If using `openthread-br/ot-posix-user-config.h` that'll be kind of duplicated and it's not so clear what `ot-posix` means in this context. Hence I'm inclined to `#include "openthread-br/user-config.h"`.